File: 3430-count-days-without-meetings/count-days-without-meetings.go
func countDays(days int, meetings [][]int) int {
    sort.Slice(meetings, func(i, j int) bool {
		return meetings[i][0] < meetings[j][0];
	})

	cntDays := meetings[0][0] - 1;
	end := meetings[0][1];

	for indx := 1; indx < len(meetings); indx++ {
		start := meetings[indx][0];

		if start > end {
			cntDays += start - end - 1;
		}
        
		if meetings[indx][1] > end {
			end = meetings[indx][1];
		}
	}

	if end < days {
		return days - end + cntDays;
	}

	return cntDays;
}

File: 3430-count-days-without-meetings/count-days-without-meetings.java
class Solution {
    public int countDays(int days, int[][] meetings) {
        int freeDays = days;
        TreeMap<Integer, Integer> meetingDays = new TreeMap<>();
        
        for (int[] meeting : meetings) {
            int start = meeting[0];
            int end = meeting[1];
            int overlapDays = 0;

            Map.Entry<Integer, Integer> previousMeetingDays = meetingDays.floorEntry(start);
            if (previousMeetingDays != null && previousMeetingDays.getValue() >= start - 1) {
                if (previousMeetingDays.getValue() >= end) continue;
                overlapDays = previousMeetingDays.getValue() - previousMeetingDays.getKey() + 1;
                start = previousMeetingDays.getKey();
            }

            Map.Entry<Integer, Integer> nextMeetingDays = meetingDays.ceilingEntry(start + 1);
            while (nextMeetingDays != null && nextMeetingDays.getKey() <= end + 1) {
                meetingDays.remove(nextMeetingDays.getKey());
                overlapDays += nextMeetingDays.getValue() - nextMeetingDays.getKey() + 1;

                if (nextMeetingDays.getValue() >= end) {
                    end = nextMeetingDays.getValue();
                    break;
                }

                nextMeetingDays = meetingDays.ceilingEntry(start + 1);
            }

            meetingDays.put(start, end);
            freeDays -= (end - start + 1) - overlapDays;

            if (freeDays == 0) break;
        }

        return freeDays;
    }
}